app/models/search/work_creator_indexer.rb

```diff
@@ -0,0 +1,39 @@
+# A class for reindexing private work creator info (info that should not be
+# available during normal searches).
+class WorkCreatorIndexer < Indexer
+  def self.klass
+    "Work"
+  end
+
+  def self.mapping
+    WorkIndexer.mapping
+  end
+
+  def routing_info(id)
+    {
+      "_index" => index_name,
+      "_type" => document_type,
+      "_id" => document_id(id),
+      "routing" => parent_id(id, nil)
+    }
+  end
+
+  def document_id(id)
+    "#{id}-creator"
+  end
+
+  def parent_id(id, _object)
+    id
+  end
+
+  def document(object)
```

Comment:
To check my understanding, every work should have this child document indexed, but the child document is only used when `viewing_own_collected_works_page?` / on works that belong to at least one collection?

Reply:
Yeah, that's the idea.